File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int,int> pos;
        for (int i =0; i<nums.size();++i ){
            if(pos.find(target - nums[i]) != pos.end()){
                res.push_back(i);
                res.push_back(pos[target-nums[i]]);
                break;
            }
            pos[nums[i]] = i;
        }
        return res;
    }
    

};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        pos = {}

        for i in range(len(nums)):
            complement   = target - nums[i]
            if complement in pos:
                return [pos[complement],i]
            pos[nums[i]] = i
        
        return []